### PR TITLE
AI docs: Updated links and notifications

### PR DESCRIPTION
**Pull request by aagonzales**
Closes https://github.com/carbon-design-system/carbon/issues/16757
Previously closed PR for ref: https://github.com/carbon-design-system/carbon-website/pull/4149

Additional change: 
Previously, we added a "coming soon" message about the new [Carbon for AI](https://carbondesignsystem.com/guidelines/carbon-for-ai/) guidelines page. I've updated that statement to now point directly to the new guidelines page.

#### Changelog

**Changed**

- Updated any instance of the word "slug" to "AI label"
- Updated "coming soon" message with the direct link to [Carbon for AI](https://carbondesignsystem.com/guidelines/carbon-for-ai/) guidelines page. 
- Updated live demos for all components from "slug" to "AI label"
- Added live demo for AI label component

**Testing**
- Make sure there are no more mentions of the word "slug" on the pages ~EXCEPT for in the Live Demo section. Dev will need to update those after the 11.64.0 release is published.~
- Make sure all AI coming soon messages have been removed/updated.
- Note the storybook links in the notification won't work until the next release.

### CODE REVIEW

**Review by laurenmrice — APPROVED** [review]
Looks good to me! No slugs. ✅

File: src/pages/components/data-table/usage.mdx [hunk 1]
```diff
@@ -654,16 +657,16 @@ present in all aspects of the data table.
...
 
 If only some data table cells have content generated by AI, then those cells
-have an in-line AI slug next to them. These cells do not get any AI layering.
+have an in-line AI label next to them. These cells do not get any AI layering.
 
 <Row>
 <Column colLg={12}>
 
-![Example of a data table cell generated by AI represneted by the in-line AI slug](images/usage-ai-presence-inline-datatable.png)
+![Example of a data table cell generated by AI represneted by the in-line AI label](images/usage-ai-presence-inline-datatable.png)
```
Comment (emyarod) [inline src/pages/components/data-table/usage.mdx:665]:
```suggestion
![Example of a data table cell generated by AI represented by the in-line AI label](images/usage-ai-presence-inline-datatable.png)
```

File: src/pages/components/data-table/usage.mdx [hunk 2]
```diff
@@ -654,16 +657,16 @@ present in all aspects of the data table.
...
 
 <Caption>
   Example of a data table cell generated by AI represneted by the in-line AI
```
Comment (emyarod) [inline src/pages/components/data-table/usage.mdx:668]:
```suggestion
  Example of a data table cell generated by AI represented by the in-line AI
```

File: src/pages/components/dropdown/usage.mdx [hunk 3]
```diff
@@ -853,12 +854,13 @@ modifier for dropdown and does not have filtering functionality._
 ## AI presence
 
 Dropdown and its varinats have a modification that takes on the AI visual
```
Comment (emyarod) [inline src/pages/components/dropdown/usage.mdx:856]:
```suggestion
Dropdown and its variants have a modification that takes on the AI visual
```

**Review by emyarod — APPROVED** [review]
looks good to me! just found a few existing typos that we can fix

File: src/pages/components/data-table/usage.mdx (hunk 1 above)
Comment (aagonzales) [reply src/pages/components/data-table/usage.mdx]: Thanks for catching those @emyarod!

**Review by alisonjoseph — APPROVED** [review]

**Pull request merged**